src/java.base/share/classes/jdk/internal/vm/annotation/Stable.java

```diff
@@ -28,56 +28,191 @@
- * optimizations of code depending on such a value (such as constant folding).
+ * There are special times early in application startup when constant
+ * folding of stable variables is disabled.  Specifically, the VM does
+ * not observe the effects of any {@code @Stable} annotation until
+ * after an AOT cache assembly phase (if any) has run to completion.
+ * Therefore, during these early times, a stable variable can be
+ * changed in any way, just like a regular mutable variable (field or
+ * array component).  If a field is annotated {@code @Stable} but is
+ * treated in this way during AOT cache assembly, that fact must be
+ * clearly stated as a warning on the field declaration.  If there is
+ * no such warning, maintainers can ignore this edge case.
+ * <p>
+ * In order to assist refactoring between {@code final} and
+ * {@code @Stable} field declarations, the Java Memory Model
+ * <em>freeze</em> operation is applied to both kinds of fields, when
+ * the assignment occurs in a class or object initializer (i.e.,
+ * static initialization code in {@code <clinit>} or constructor code
+ * in {@code <init>}).  The freezing of a final or stable field is
+ * (currently) triggered only when an actual assignment occurs, directly
+ * from the initializer method ({@code <clinit>} or {@code <init>}).
+ * It is implemented in HotSpot by an appropriate memory barrier
+ * instruction at the return point of the initializer method.  In this
+ * way, any non-null (or non-zero) value stored to a stable variable
+ * (either field or array component) will appear without races to any
+ * user of the class or object that has been initialized.
+ * <p>
+ * There is no such freeze operation applied to stable field stores in
+ * any other context.  This implies that a constructor may choose to
+ * initialize a stable variable, rather than "leaving it for later",
+ * and that initial will be safely published, as if the field were
+ * {@code final}.  The stored value may (or may not) contain
+ * additional stable variables, not yet initialized.  Note that if a
+ * stable variable is written outside of the code of a constructor (or
+ * class initializer), then data races are possible, just the same as
+ * if there were no {@code @Stable} annotation, and the field was a
+ * regular mutable field.  In fact, the usual case for lazily
+ * evaluated data structures is to assign to stable variables much
+ * later than the enclosing data structure is created.  This means
+ * that racing reads and writes might observe nulls (or primitive
+ * zeroes) as well as non-default values.
+ * <p>
+ * Therefore, most code which reads stable variables should not assume
+ * that the value has been set, and should dynamically test for a null
+ * (or zero) value.  Code which cannot prove proper ordering of
+ * initialization may use stable variables without performing the null
+ * (or zero) test.  Code which omits the null (or zero) test should be
```

Comment:
```
 * (or zero) value.  Code which cannot prove proper ordering of
 * initialization may use stable variables without performing the null
 * (or zero) test.  Code which omits the null (or zero) test should be
```
Do we want to say "may use without null check" or is there a missing "should not" in that sentence?